__mocks__/bin-wrapper.js

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-use-before-define */
+const src = jest.fn(() => binWrapperObject);
```

Comment:
Wouldn't binWrapperObject throw a ReferenceError at this point since `const` variables are not hoisted and is binWrapperObject is declared below? 

Or I guess it is not referenced until this callback is called... in which case binWrapperObject is declared in the parent scope? 🤔

Reply:
Honestly not sure at all. I just tried stuff until it worked for this one, which was tricky since I needed the mutually recursive definitions. In any case, the mock seems to behave as expected. 🤷🏻‍♂️